src/NftDetails/NftDetails.tsx

```diff
@@ -1,43 +1,48 @@
-import {useNavigation, useRoute} from '@react-navigation/native'
-import React, {useEffect, useState} from 'react'
+import {RouteProp, useNavigation, useRoute} from '@react-navigation/native'
+import React, {ReactNode, useMemo, useState} from 'react'
+import {defineMessages, useIntl} from 'react-intl'
 import {Image, StyleSheet, TouchableOpacity, View} from 'react-native'
 import {ScrollView} from 'react-native-gesture-handler'
 
-import {CopyButton, FadeIn, OfflineBanner, Spacer, StatusBar, Text} from '../components'
+import {getPrettyDate} from '../../tests/helpers/utils'
+import {CopyButton, FadeIn, Icon, Link, Spacer, StatusBar, Text} from '../components'
 import {Tab, TabPanel, TabPanels, Tabs} from '../components/Tabs'
-import {useNfts} from '../hooks'
-import {NftDetailsNavigation} from '../navigation'
+import {useNft, useTransactionInfos} from '../hooks'
+import {getAssetFingerprint} from '../legacy/format'
+import {NftDetailsNavigation, NftRoutes} from '../navigation'
 import {useSelectedWallet} from '../SelectedWallet'
+import {COLORS} from '../theme'
 
-enum VIEW_TABS {
-  OVERVIEW = 'OVERVIEW',
-  METADATA = 'METADATA',
-}
-
-type Params = {id: string}
+type ViewTabs = 'overview' | 'metadata'
 
 export const NftDetails = () => {
+  const {id} = useRoute<RouteProp<NftRoutes, 'nft-details'>>().params
+  const strings = useStrings()
   const wallet = useSelectedWallet()
-  const {nfts} = useNfts(wallet)
+  const transactionsInfo = useTransactionInfos(wallet)
 
   const navigation = useNavigation<NftDetailsNavigation>()
-  const [activeTab, setActiveTab] = useState<VIEW_TABS>(VIEW_TABS.OVERVIEW)
-  const {id} = useRoute().params as Params
-  const nft = nfts.find((nft) => nft.id === id)
+  const [activeTab, setActiveTab] = useState<ViewTabs>('overview')
+  const nft = useNft(wallet, {id})
   const stringifiedMetadata = JSON.stringify(nft, undefined, 2)
+  const fingerprint = useMemo(
+    () => (nft !== null ? getAssetFingerprint(nft.metadata.policyId, nft.metadata.assetNameHex) : null),
```

Comment:
under what circumstances can nft be null?

Reply:
I don't think there's a real case where nft would be null, but TypeScript expects the type to be null | NFT, so I added an additional check to be on the safe side.

useMemo just because we're using a "3rd" party library to compute the fingerprint. And the lib uses buffers and multiple times converts the data. Nothing to worry about, but speaking about performance, it will be easier to make a dependency comparison than to recompute the values

Reply:
if it's slow, why doesn't it use promises?

Reply:
can you push it into it's own component and memoize the component instead?

Reply:
where does it become the `Nft | null`?

Reply:
Here 
```
export const useNft = (wallet: YoroiWallet, {id}: {id?: string} = {}): YoroiNFT | null => {
  const {nfts} = useNfts(wallet)
  return nfts.find((nft) => nft.id === id) ?? null
}
```

Reply:
throw an error instead
right a story for not found